### DISCUSSION TITLE
Combine Query and Vector search

### DISCUSSION THREAD

**Discussion by Dananale** [milvus-io/milvus#35986] [Q&A and General discussion]
Hello, 
I use Milvus with a schema that stores Slack threads.
The schema consists of Date, Name and The conversation text which is the vectorized (and some other information)
My question is can I query and search combined.
For example, I want a summary of all threads initiated by a certain user between certain dates. 
Another example of a query would be: I want to search a certain phrase, using the vector field, for all conversation between certain dates. 
I can do this in multiple steps, but thought perhaps it can be done also using Milvus API

**Comment by xiaofan-luan** [comment]
for sure you can do it. filter search might be what you are looking for https://milvus.io/docs/single-vector-search.md#Filtered-search

With the filtering search you can get result related to your vector but restricted to certain filtering conditions

**Reply by Dananale** [reply]
Thank you! I think this can do the job.
In the documentation, the data filed supposed to be: (Union[List[list], list]): The vector/vectors to search. 
what does exactly this mean? Is it the embedding presentation of the query?

**Reply by yhmo** [reply]
@Dananale 
The legal value type of "data" is List[list] or list, which represents a vector or multiple vectors to be searched in this collection. Yes, we can say "it is the embedding presentation of the query".
```
def search(
        self,
        collection_name: str,
        data: Union[List[list], list],
        ......
)
```

Valid input:
- a list of list[float] like [[1, 2, 3, ...], [2, 3, 4, ...], ......]
- a list of numpy.ndarray
- a numpy.ndarray
